### PR TITLE
SSH over SSM and flyway version input

### PR DESCRIPTION
**Pull request by mi-kas**
## What/Why/How?

- Make flyway version configurable via an optional input parameter (`FLYWAY_VERSION`)
- Use ssh over ssm to connect to the bastion host
- Use `InstanceId` as identifier for instance id query

### CODE REVIEW

**Comment by pmaedel** [comment]
@dasmarcomoreira please take the lead here :)

**Comment by mi-kas** [comment]
Ping @dasmarcomoreira

File: .github/workflows/migration.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Stage Migration
+name: DB Migration
```
Comment (dasmarcomoreira) [inline .github/workflows/migration.yaml:1]: Why DB Migration ? Every shared workflow as the following pattern "Stage <Something>". Any reason for change?

**Review by dasmarcomoreira — DISMISSED** [review]
lgtm, sorry about the delay

File: .github/workflows/migration.yaml (hunk 1 above)
Comment (mi-kas) [reply .github/workflows/migration.yaml:1]: Some teams have a "stage" environment so I thought it might make sense to improve the naming here.

**Comment by mi-kas** [comment]
@dasmarcomoreira I am missing the permissions to merge the PR, so feel free to do it if everything looks good to you

File: .github/workflows/migration.yaml (hunk 1 above)
Comment (dasmarcomoreira) [reply .github/workflows/migration.yaml:1]: If we want to improve naming we must do it across workflows. I am apologist of: bad pattern better than no pattern :D

Comment (dasmarcomoreira) [reply .github/workflows/migration.yaml:1]: Would it make sense from your side:  Stage Database Migration ?

Comment (mi-kas) [reply .github/workflows/migration.yaml:1]: Yes, that would be perfect 👍 Will change the naming tomorrow then

**Review by pmaedel — APPROVED** [review]

**Comment by pmaedel** [comment]
@kimakunc your review is required as you are a codeowner for files changed here

**Comment by dasmarcomoreira** [comment]
No need for a code owner to review anymore, merge can now be done by you @mi-kas

File: .github/workflows/migration.yaml (hunk 1 above)
Comment (mi-kas) [reply .github/workflows/migration.yaml]: Improved the naming now and would need a new review approval @dasmarcomoreira

**Review by dasmarcomoreira — APPROVED** [review]

**Pull request merged**